packages/vite/src/client/client.ts

```diff
@@ -140,20 +142,36 @@ const hmrClient = new HMRClient(
   },
   transport,
   async function importUpdatedModule({
+    url,
     acceptedPath,
     timestamp,
     explicitImportRequired,
     isWithinCircularImport,
   }) {
-    const [acceptedPathWithoutQuery, query] = acceptedPath.split(`?`)
-    const importPromise = import(
-      /* @vite-ignore */
-      base +
-        acceptedPathWithoutQuery.slice(1) +
-        `?${explicitImportRequired ? 'import&' : ''}t=${timestamp}${
-          query ? `&${query}` : ''
-        }`
-    )
+    function importModuleWithFullBundleMode() {
+      const importPromise = import(base + url)
+      return importPromise.then(() =>
+        // @ts-expect-error globalThis.__rolldown_runtime__
```

Comment:
In the future, I think it's good to have the types exposed from rolldown so that we can use it like:
```ts
declare var __rolldown_runtime__: RolldownHmrRuntime
```

Reply:
Yeah.